Assets/MixedRealityToolkit.Providers/UnityAR/UnityARCameraSettings.cs

```diff
@@ -0,0 +1,255 @@
+            }
+            else
+            {
+                CameraCache.Main.clearFlags = cameraProfile.CameraClearFlagsTransparentDisplay;
+                CameraCache.Main.backgroundColor = cameraProfile.BackgroundColorTransparentDisplay;
+                CameraCache.Main.nearClipPlane = cameraProfile.NearClipPlaneTransparentDisplay;
+                CameraCache.Main.farClipPlane = cameraProfile.FarClipPlaneTransparentDisplay;
+                QualitySettings.SetQualityLevel(cameraProfile.TransparentQualityLevel, false);
+            }
+        }
+
+        #endregion IMixedRealityCameraSettings
+
+        /// <summary>
+        /// The profile used to configure the camera.
+        /// </summary>
+        public UnityARCameraSettingsProfile SettingsProfile
+        {
+            get
+            {
+                return ConfigurationProfile as UnityARCameraSettingsProfile;
+            }
+        }
+
+        bool isInitialized = false;
+
+        private GameObject arSessionObject = null;
+        private bool preExistingArSessionObject = false;
+        private ARSession arSession = null;
+
+        private GameObject arSessionOriginObject = null;
+        private bool preExistingArSessionOriginObject = false;
+        private ARSessionOrigin arSessionOrigin = null;
+
+        private ARCameraManager arCameraManager = null;
+        private ARCameraBackground arCameraBackground = null;
+        private ARInputManager arInputManager = null;
+        private TrackedPoseDriver trackedPoseDriver = null;
+
+        /// <summary>
+        /// <summary>
+        /// Examines the scene to determine if AR Foundation components are present.
+        /// </summary>
+        private void FindARFoundationComponents()
+        {
+            arSessionObject = GameObject.Find("AR Session");
+            preExistingArSessionObject = (arSessionObject != null);
+            arSessionOriginObject = GameObject.Find("AR Session Origin");
+            preExistingArSessionOriginObject = (arSessionOriginObject != null);
+        }
+
+        public override async void Initialize()
+        {
+            base.Initialize();
+
+            ARSessionState sessionState = (ARSessionState)(await ARSession.CheckAvailability());
+            if (ARSessionState.Ready > sessionState)
+            {
+                Debug.LogError("Unable to initialize the Unity AR Camera Settings provider. Device support for AR Foundation was not detected.");
+                isInitialized = true;
+            }
+        }
+
+        public override void Enable()
+        {
+            base.Enable();
+            
+            if (!isInitialized)
+            {
+                InitializeARFoundation();
+            }
+        }
+
+        public override void Destroy()
+        {
+            UninitializeARFoundation();
+
+            base.Destroy();
+        }
+
+        /// <summary>
+        /// Initialize AR Foundation components.
```

Comment:
Maybe elaborate what components are being added and why

Reply:
added a brief remarks section